src/kernels/helpers.ts

```diff
@@ -136,13 +135,13 @@ export function getInterpreterHashInMetadata(
         return (notebookMetadata as any).interpreter.hash;
     }
 }
-export function findPreferredKernel(
+export function rankKernelsImpl(
```

Comment:
nit: can we remove "Impl" from the function names? Typically used when overloading for the private implemenation, but this and findExactMatch are exported

Reply:
Yeah, that's more than fair. Impl has specific meaning. Hmm, thinking on what I would name a helper function providing much of the functionality for function of the same name.